opensearch_py_ml/ml_models/sentencetransformermodel.py

```diff
@@ -163,12 +188,30 @@ def train(
                     num_processes=num_processes,
                 )
             else:
-                subprocess.run(
-                    [
-                        "accelerate launch self.train_model(train_examples, self.model_id, output_model_path, output_model_name,use_accelerate,learning_rate,num_epochs,verbose)"
-                    ]
-                )
-        else:
+                try:
+                    subprocess.Popen(
```

Comment:
> I am curious why changing `subprocess.run` to `subprocess.Popen`? Normally, 'The recommended approach to invoking subprocesses is to use the [run()](https://docs.python.org/3/library/subprocess.html#subprocess.run) function for all use cases it can handle.'

[subprocess.run()](https://docs.python.org/3/library/subprocess.html#subprocess.run) [was added in Python 3.5](https://docs.python.org/3/whatsnew/3.5.html#subprocess) as a simplification over [subprocess.Popen](https://docs.python.org/3/library/subprocess.html#subprocess.Popen) when you just want to execute a command and wait until it finishes, but you don't want to do anything else in the mean time. For other cases, you still need to use subprocess.Popen.

The main difference is that subprocess.run() executes a command and waits for it to finish, while with subprocess.Popen you can continue doing your stuff while the process finishes and then just repeatedly call [Popen.communicate()](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate) yourself to pass and receive data to your process. Secondly, subprocess.run() returns [subprocess.CompletedProcess](https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess).

You could say with `Popen` you can achieve some sort of asynchronous programming which I tried to do to generate some output. But in the end it didn't improve the code so I reverted now to the simplest approach.